### PR TITLE
[multikueue] Batch remote workload reconcile events.

### PR DESCRIPTION
**Pull request by trasc**
#### What type of PR is this?

/kind feature

#### What this PR does / why we need it:
[multikueue] Batch remote workload reconcile events.
#### Which issue(s) this PR fixes:

Relates to #693 

#### Does this PR introduce a user-facing change?

```release-note
[multikueue] Batch remote workload reconcile events.
```

### CODE REVIEW

File: pkg/controller/admissionchecks/multikueue/workload.go [hunk 1]
```diff
@@ -458,24 +459,25 @@ func (w *wlReconciler) Generic(_ event.GenericEvent) bool {
...
 
 func (w *wlReconciler) setupWithManager(mgr ctrl.Manager) error {
 	syncHndl := handler.Funcs{
 		GenericFunc: func(_ context.Context, e event.GenericEvent, q workqueue.RateLimitingInterface) {
-			q.Add(reconcile.Request{NamespacedName: types.NamespacedName{
+			q.AddAfter(reconcile.Request{NamespacedName: types.NamespacedName{
 				Namespace: e.Object.GetNamespace(),
 				Name:      e.Object.GetName(),
-			}})
+			}}, w.eventsBatchPeriod)
```
[the remark this answers is not in the document]
Comment (trasc) [reply pkg/controller/admissionchecks/multikueue/workload.go:480]: More like not trigger more the one reconcile for the same workload within as `eventsBatchPeriod` window

File: pkg/controller/admissionchecks/multikueue/controllers.go [hunk 2]
```diff
@@ -60,11 +63,20 @@ func WithWorkerLostTimeout(d time.Duration) SetupOption {
 	}
 }
 
+// WithEventBatchPeriod - sets the delay used when adding remote triggered
+// events to the workload's reconcile queue.
+func WithEventBatchPeriod(d time.Duration) SetupOption {
```
Comment (mimowo) [inline pkg/controller/admissionchecks/multikueue/controllers.go:68]:
```suggestion
// WithEventsBatchPeriod - sets the delay used when adding remote triggered
// events to the workload's reconcile queue.
func WithEventsBatchPeriod(d time.Duration) SetupOption {
```
```suggestion
// WithEventBatchPeriod - sets the delay used when adding remote triggered
// events to the workload's reconcile queue.
func WithEventBatchPeriod(d time.Duration) SetupOption {
```
for naming consistency.

**Review by mimowo — COMMENTED** [review]
Generally lgtm.  I just doubt we benefit from parametrizing it for tests if the batch period is just 1s. 

We could just use the constant as in other use-cases: https://github.com/search?q=repo%3Akubernetes-sigs%2Fkueue+constants.UpdatesBatchPeriod&type=code.

File: pkg/controller/admissionchecks/multikueue/workload_test.go [hunk 3]
```diff
@@ -1050,7 +1050,7 @@ func TestWlReconcile(t *testing.T) {
 			}
 
 			helper, _ := newMultiKueueStoreHelper(managerClient)
-			reconciler := newWlReconciler(managerClient, helper, cRec, defaultOrigin, defaultWorkerLostTimeout)
+			reconciler := newWlReconciler(managerClient, helper, cRec, defaultOrigin, defaultWorkerLostTimeout, time.Second)
```
Comment (mimowo) [inline pkg/controller/admissionchecks/multikueue/workload_test.go:1053]: I would be ok not to parametrize it. but if you do, then let's use 100ms to get (slightly) better runtimes.

Comment (trasc) [reply pkg/controller/admissionchecks/multikueue/workload_test.go:1053]: The value is not relevant in this test.

**Comment by mimowo** [comment]
I'm still in doubt what is the benefit of making it parametrized, but it is non-blocking.
/lgtm
/approve

**Comment by alculquicondor** [comment]
/release-note-edit
```release-note
Multikueue: Batch reconcile events for remote workloads.
```

**Pull request merged**